rfcs/text/0428-type-inheritance-and-forking.md

```diff
@@ -0,0 +1,952 @@
+           ├───Person─────Employee (subtype)
+       age◄┘              │
+                          │
+occupation◄───────────────┘
+```
+
+**An example of _compatible_, overlapping properties**:
+
+- Supertype `Person` contains the required properties `name` and `age`
+- Supertype `Superhero` contains the required properties `superpower` and `name`
+
+In this example, `name` overlaps as a required property in both supertypes. Compatibility of overlapping properties is defined in the [versioning RFC's definition of "compatible"](./0408-versioning-types.md#determining-type-compatibility).
+
+```txt
+              (supertypes)
+superpower◄────Superhero──┐
+               │          │
+      name◄────┤          │
+               │          │
+       age◄────Person─────Employee (subtype)
+                          │
+occupation◄───────────────┘
+
+
+```
+
+**An example of _incompatible_, overlapping properties**:
+
+- Supertype `Person` contains the required properties `name` and `age`
+- Supertype `Superhero` contains the required property `superpower` and an _array_ of `name`s
+
+In this example, the array of `name`s on the `Superhero` type would not be compatible with the required `name` property of `Person`, which means that the two types cannot be supertypes together.
+
+## The `additionalProperties` problem
+
+In the proposed [Versioning RFC](./0408-versioning-types.md) for the type system, having `{ "additionalProperties": false }` for all schemas is an assumption made for determining type compatibility, which means that any supertype will not validate against a subtype that adds properties if it receives all properties of a subtype instance. For example, if we supply the `Employee` instance from above to a `Person`, it will receive properties that are considered `additionalProperties` (the `occupation` property is not present on `Person`).
+
+The assumption that we can select/project parts of a subtype that make up a supertype is essential for keeping strictness in JSON Schemas.
+
+We propose slight modifications to how `{ "additionalProperties": true }` and `{ "unevaluatedProperties": false }` behave and may be used within our type extension system to let supertypes keep strictness while allowing composition. This modification is a semantic change and does not change any syntax.
+
+Concrete examples of how JSON Schema breaks with these validation constraints are shown in the [Reference-level explanation](#problems-with-unevaluatedproperties).
+
+## Defining extended entity types
+
+Extended types will be defined with conventional JSON Schema syntax: the `allOf` keyword. An entity type can extend another entity type by adding a versioned URI reference to the root-level `allOf` array.
+A [versioned URI](https://github.com/blockprotocol/blockprotocol/blob/main/rfcs/text/0408-versioning-types.md#type-uris) is used so that subtypes aren't automatically updated (and potentially invalidated) when the supertype is updated.
```

Comment:
```suggestion
A [versioned URL](https://github.com/blockprotocol/blockprotocol/blob/main/rfcs/text/0408-versioning-types.md#type-uris) is used so that subtypes aren't automatically updated (and potentially invalidated) when the supertype is updated.
```

Reply:
Swept through URI in 3615bb1848ac4250f017197cfc1aa2536f424df0 :)

Reply:
Thanks!